File: BE/src/post/dto/postCreate.dto.ts
import { IsBoolean, IsString, MaxLength } from 'class-validator';
import { IsPriceCorrect } from '../../common/decorator/price.decorator';

export class PostCreateDto {
  @IsString()
  @MaxLength(100)
  title: string;

  @IsString()
  description: string;

  @IsPriceCorrect('is_request')
  price: number;

  @IsBoolean()
  is_request: boolean;

  @IsString()
  start_date: string;

  @IsString()
  end_date: string;
}

File: iOS/Village/Village/Domain/Network/PostCreateDTO.swift
//
//  PostCreateDTO.swift
//  Village
//
//  Created by 조성민 on 11/24/23.
//

import Foundation

struct PostCreateInfo: Codable {
    
    let postInfo: PostInfoDTO
    let images: [ImageDTO]
    
}

struct PostInfoDTO: Codable {
    
    let title: String
    let description: String
    let price: Int?
    let isRequest: Bool
    let startDate: String
    let endDate: String
    
    enum CodingKeys: String, CodingKey {
        case title
        case price
        case description
        case isRequest = "is_request"
        case startDate = "start_date"
        case endDate = "end_date"
    }
}

struct ImageDTO: Codable {
    let fileName: String
    let type: String
    let data: Data
    
    enum CodingKeys: String, CodingKey {
        case fileName = "filename"
        case type
        case data
    }
}
